lib/timeout.rb

```diff
@@ -32,19 +32,21 @@ class Error < RuntimeError
     def self.catch(*args)
       exc = new(*args)
       exc.instance_variable_set(:@thread, Thread.current)
-      ::Kernel.catch(exc) {yield exc}
+      catch_value = Object.new
+      exc.instance_variable_set(:@catch_value, catch_value)
+      ::Kernel.catch(catch_value) {yield exc}
```

Comment:
Is the different object needed?

```suggestion
      exc.instance_variable_set(:@catch_value, exc)
      ::Kernel.catch(exc) {yield exc}
```

Reply:
I'll try that approach, it would be good to avoid the object allocation.